runtime/src/account_info.rs

```diff
@@ -63,22 +63,25 @@ pub struct AccountInfo {
 
     /// needed to track shrink candidacy in bytes. Used to update the number
     /// of alive bytes in an AppendVec as newer slots purge outdated entries
-    /// Note that highest bit is used for ZERO_LAMPORT_BIT
-    stored_size: StoredSize,
+    /// Note that highest bits are used by ALL_BITS
+    /// So, 'stored_size' is 'stored_size_mask' with ALL_BITS masked out.
+    stored_size_mask: StoredSize,
 }
 
-/// presence of this bit in stored_size indicates this account info references an account with zero lamports
+/// These bits can be present in stored_size_mask to indicate additional info about the AccountInfo
 const ZERO_LAMPORT_BIT: StoredSize = 1 << (StoredSize::BITS - 1);
+const CACHED_STORE_ID_BIT: StoredSize = 1 << (StoredSize::BITS - 2);
+const ALL_BITS: StoredSize = ZERO_LAMPORT_BIT | CACHED_STORE_ID_BIT;
```

Comment:
nit: I would change BIT -> FLAG

Reply:
nit `CACHED_STORE_ID_BIT` -> `IS_CACHED_STORE_ID_BIT`

Maybe a comment here as well similar to the one for `ZERO_LAMPORT_BIT` about what this flag indicates